src/nexgen/tools/ED_tools.py

```diff
@@ -155,6 +155,30 @@ def get_data_collection_date(self) -> str:
             return collection_date
 
 
+def extract_start_time_from_master(master: Path | str) -> datetime:
+    """
+    Extracts `data_collection_data` from the master file
+
+    Args:
+        master (Path | str): Path to Singla master file.
+
+    Returns:
+        data collection date: Datetime object.
+    """
+    import logging
```

Comment:
I realise now that it probably would be better to now the logging defined just once in the script outside of both functions (since we're going to needed it in multiple places). Happy to do that at a later time in separate issue though.